### ISSUE TITLE
_mark_immutable does not handle non-ASCII strings

### ISSUE THREAD

**Issue by lucaswiman** [ppannuto/python-titlecase#26]
Here is a simplified reproduction of an issue I'm seeing while trying to upgrade `titlecase`:
```python
>>> from titlecase import titlecase
>>> titlecase(u'resumé')
u'Resum\xe9'
>>> titlecase(u'resumé', callback=lambda x: x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lucaswiman/counsyl/website2/vendor/venv/lib/python2.7/site-packages/titlecase/__init__.py", line 72, in titlecase
    new_word = callback(word, all_caps=all_caps)
TypeError: <lambda>() got an unexpected keyword argument 'all_caps'
>>> titlecase(u'resumé', callback=lambda word, **kw: word)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lucaswiman/counsyl/website2/vendor/venv/lib/python2.7/site-packages/titlecase/__init__.py", line 76, in titlecase
    tc_line.append(_mark_immutable(new_word))
  File "/Users/lucaswiman/counsyl/website2/vendor/venv/lib/python2.7/site-packages/titlecase/__init__.py", line 38, in _mark_immutable
    return ImmutableString(text)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 5: ordinal not in range(128)
```

I'll try to submit a pull request for this at some point in the near future.

**Comment by ppannuto** [comment]
Looks like this is a Python2 issue:
```
$ python3
Python 3.6.1 (default, Mar 23 2017, 16:49:06)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import titlecase
>>> titlecase.titlecase(u'resumé')
'Resumé'
>>> titlecase.titlecase(u'resumé', callback=lambda word, **kw: word)
'resumé'

$ python2
Python 2.7.13 (default, Dec 18 2016, 07:03:39)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import titlecase as t
>>> t.titlecase(u'resumé')
u'Resum\xe9'
>>> help(t.titlecase)

>>> t.titlecase(u'resumé', callback=lambda word, **kw: word)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "titlecase/__init__.py", line 75, in titlecase
    tc_line.append(_mark_immutable(new_word))
  File "titlecase/__init__.py", line 37, in _mark_immutable
    return ImmutableString(text)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 5: ordinal not in range(128)
```

I'm personally pretty disinclined to put any effort into supporting Py2k in 2017. I'm happy to take a PR to fix this, but for now I'm probably going to leave it as a wontfix

**Comment by lucaswiman** [comment]
> I'm happy to take a PR to fix this, but for now I'm probably going to leave it as a wontfix

Thanks for commenting; I'd forgotten to submit a patch for this. I wasn't trying to get _you_ to fix it; sorry if I came off that way. Many maintainers strongly prefer filing an issue before submitting a patch, which is why I filed the issue.

> I'm personally pretty disinclined to put any effort into supporting Py2k in 2017. 

That's understandable; Python 2 can't die fast enough. The impetus for this bug came from upgrading all the dependencies in a Python 2 project as preparation to upgrade to Python 3. This is an especially annoying bug for the transition from Python 2 to Python 3. It has the backwards semantics from what you'd want (does not work for `unicode` in Python 2, but does work for bytestrings), making it hard to write compatible code.
